internal/controller/telemetry/collector_controller.go

```diff
@@ -563,6 +568,44 @@ func normalizeStringSlice(inputList []string) []string {
 	return uniqueList
 }
 
+func appendAdditionalVolumesForTenantsFileStorage(otelComonFields *otelv1beta1.OpenTelemetryCommonFields, tenants []v1alpha1.Tenant) {
+	for _, tenant := range tenants {
+		if tenant.Spec.PersistenceConfig.EnableFileStorage {
+			mountPath := storage.DetermineFileStorageDirectory(tenant.Spec.PersistenceConfig.Directory)
+			switch tenant.Spec.PersistenceConfig.VolumeSource {
+			case "hostPath":
+				otelComonFields.VolumeMounts = append(otelComonFields.VolumeMounts, corev1.VolumeMount{
+					Name:      fmt.Sprintf("buffervolume/%s", tenant.Name),
+					MountPath: mountPath,
+				})
+				otelComonFields.Volumes = append(otelComonFields.Volumes, corev1.Volume{
+					Name: fmt.Sprintf("buffervolume/%s", tenant.Name),
+					VolumeSource: corev1.VolumeSource{
+						HostPath: &corev1.HostPathVolumeSource{
+							Path: mountPath,
+							Type: utils.ToPtr(corev1.HostPathDirectoryOrCreate),
+						},
+					},
+				})
+
+			case "emptyDir":
+				otelComonFields.VolumeMounts = append(otelComonFields.VolumeMounts, corev1.VolumeMount{
+					Name:      fmt.Sprintf("buffervolume/%s", tenant.Name),
+					MountPath: mountPath,
+				})
+				otelComonFields.Volumes = append(otelComonFields.Volumes, corev1.Volume{
+					Name: fmt.Sprintf("buffervolume/%s", tenant.Name),
+					VolumeSource: corev1.VolumeSource{
+						EmptyDir: &corev1.EmptyDirVolumeSource{
```

Comment:
Not a big fan of this as this defeats the purpose of persistence. I won't block this PR just because of this, but let's give this a second thought before going forward with it.

Reply:
EmptyDir is removed for now